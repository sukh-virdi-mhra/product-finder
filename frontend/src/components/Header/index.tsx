import React from "react"
import styled from "styled-components"

const StyledH1 = styled.h1`
  color: blue;
  font-size: 80px;
`

interface HeaderProps {
  name: string
  age: number
  children: React.ReactNode
}

export default function Header(props: HeaderProps) {
  return <StyledH1>{props.children}</StyledH1> 
  
}