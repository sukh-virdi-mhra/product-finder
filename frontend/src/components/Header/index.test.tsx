import React from "react"
import { shallow, mount } from "enzyme"

import Header from "./index"

describe("Header", () => {
  it("renders shallow", () => {
    const component = shallow(
      <Header name="whatever" age={12}>
        Here is some example content
      </Header>
    )
    expect(component).toMatchSnapshot()
  }),
    it("renders mount", () => {
      const component = mount(
        <Header name="whatever" age={12}>
          Here is some example content
        </Header>
      )
      expect(component).toMatchSnapshot()
    })
})