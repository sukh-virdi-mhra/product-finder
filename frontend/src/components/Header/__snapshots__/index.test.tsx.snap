// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders mount 1`] = `
<Header
  age={12}
  name="whatever"
>
  <styled.h1>
    <h1
      className="sc-AxjAm bwvbkn"
    >
      Here is some example content
    </h1>
  </styled.h1>
</Header>
`;

exports[`Header renders shallow 1`] = `
<styled.h1>
  Here is some example content
</styled.h1>
`;
